sujeong/app.js

```diff
@@ -45,12 +47,137 @@ app.post('/users',async (req ,res,next)=>{
             email,
             profile_image,
             password
-        ) VALUES ( ?, ?, ?, ?)
+        ) VALUES ( ?, ?, ?, ?);
         `,[name, email, profile_image, password]
     );
     res.status(201).json({message:"userCreated"})
 })
 
+
+app.post('/posts',async(req, res, next)=>{
+    const {title, content, user_id,image_url} =req.body
+    
+    await appDataSource.query(
+        `INSERT INTO posts(
+            title,
+            content,
+            user_id,
+            image_url
+        ) VALUES (?,?,?,?);`
+    ,[title, content, user_id, image_url]
+    );
+    res.status(201).json({message: "postCreated"})
+})
+
+
+app.get('/posts/userId', async(req, res)=>{
+    
+    await appDataSource.query(
+    `SELECT
+        u.id as userId,
+        u.profile_image as userProfileImage,
+        p.user_id as postingId,
+        p.image_url as postingImageUrl,
+        p.content as postingContent
+    FROM users u
+    INNER JOIN posts p ON u.id = p.user_id;
+    `,(err, rows) => {
+        res.status(200).json(rows);
+    });
+    });
+
+
+app.get('/posts/:userId', async (req , res)=>{
+    const {userId} = req.params
+    
+    try {
+        const [result] = await appDataSource.query(
+        `SELECT
+            u.id userId,
+            u.profile_image userProfileImage,
+        JSON_ARRAYAGG(
+            JSON_OBJECT(
+            "postingId", p.id,
+            "postingImageUrl", p.image_url,
+            "postingContnet", p.content)  
+        ) as postings
+        FROM users u 
+        JOIN posts p ON p.user_id = u.id
+        WHERE u.id = ${userId}
+        GROUP BY u.id`
+        );
+        return res.status(200).json({ data: result });
+    } catch (err) {
+        return res.status(409).json({ error: 'invalid input' });
+    }
+    });
+
+
+app.put('/posts/:postId', async(req,res)=>{
+    const {title, content, image_url} =req.body
+    const { postId } = req.params
+    
+    try{
+    await appDataSource.query(
+        `UPDATE posts
+        SET 
+            title= ?,
+            content= ?,
+            image_url= ?
+        WHERE id= ${postId}
+            `, [title, content, image_url]
+    );
+    const [result] = await appDataSource.query(
+        `SELECT
+            u.id userId,
+            u.name userName,
+            p.id postingId,
+            p.title postingTitle,
+            p.content postingContent
+        FROM users u
+        JOIN posts p ON u.id = p.user_id
+        WHERE p.id =${postId}
+        `
+    );
+    
+    res.status(200).json({data: result});
+    }catch (err){
+        return res.status(409).json({"message": "invalid input"})
+    }
+});
+
```

Comment:
수정님 !  409 http status code는 서버 충돌이 발생함을 의미하는 코드입니다!
상황에 맞는 http status code를 다시 수정해 주세요!
```suggestion
    res.status(200).json({data: result});
    } catch (err) {
        return res.status(409).json({"message": "invalid input"})
    }
});

```

Reply:
401로 바궜습니다!